### PR TITLE
[Refactor] Clarify dependency construction for the CallComposite

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
`DependencyContainer.swift` contained at type-erased dictionary of objects, primarily used by the CallComposite class.
The pattern of being able to `register()` and `resolve()` objects at run-time is convenient for applications where they will be changed at run-time, but this is not the case with our library.

`DependencyContainer.swift` provided a means to do a lot of object construction, which is then handed back to the calling class.

This PR constructs all dependencies for the CallComposite internally and owns those objects.

As a result, we have
* Reduced complexity in the code
* Reduced the number of test cases 
* Maintained type-checking at compile time
* Clarified object usage in the code

The chat code has a similar PR:
https://github.com/Azure/communication-ui-library-ios/pull/494

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

**Review by palatter — APPROVED** [review]
This looks good! Thanks for simplifying this!

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Presentation/Setup/CallCompositeTests.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class CallCompositeTests: XCTestCase {
+    func test_verify_construction() {
+        let callComposite = CallComposite()
+
+        XCTAssertNotNil(callComposite)
+    }
+
+    func test_verify_launch() throws {
+
+        let sampleToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMyNTAzNjgwMDAwfQ.9i7FNNHHJT8cOzo-yrAUJyBSfJ-tPPk2emcHavOEpWc"
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Presentation/Setup/CallCompositeTests.swift:21]: I'm not sure its a good idea to check in a token into the repo. Maybe we can inject in the AppConfig

Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Presentation/Setup/CallCompositeTests.swift:21]: I just moved it. It's in the repo already.

Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Presentation/Setup/CallCompositeTests.swift:21]: There's nothing in the token; have a look at https://jwt.io to evaluate.

**Review by JoshuaLai — APPROVED** [review]

**Pull request merged**